pkg/cri/server/netapi_adaptor.go

```diff
@@ -0,0 +1,224 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"reflect"
+	"runtime"
+	"strings"
+
+	"github.com/containerd/containerd/pkg/net"
+	"github.com/containerd/containerd/pkg/net/compat"
+	"github.com/containerd/go-cni"
+)
+
+// cniAdaptor is created to adapt the APIs of network plugin to their
+// counterparts in go-cni.
+type cniAdaptor struct {
```

Comment:
The adapter is for both `g` and `c`. Isn't cni.CNI and compat.CNI mostly compatible?